app/API/v1/views/user_registration.py

```diff
@@ -28,39 +28,51 @@ def allowed_file(filename):
 
 class Users(Resource):
   '''User registration'''
   def post(self):
     arguments = user_registration.parse_args()
-    first_name = arguments['first_name']
-    last_name = arguments['last_name']
-    other_name = arguments['other_name']
+    firstname = arguments['firstname'].strip()
+    lastname = arguments['lastname'].strip()
+    othername = arguments['othername']
     email = arguments['email']
-    phone_number = arguments['phone_number']
-    passport = arguments['passport']
+    phonenumber = arguments['phonenumber']
+    passporturl = arguments['passporturl']
     password = arguments['password']
   
-    if passport:
-      passport_url = secure_filename(passport.filename)
-      passport.save(os.path.join(app.config['UPLOAD_FOLDER'], passport_url))
-    if not first_name or not last_name or not other_name or not email or not phone_number or not passport_url or not password:
-      return {"message": "Please provide first_name or last_name or other_name or email or phone_number or passport_url"}, 400
-    if len(password) < 6:
-      return {"message": "Password length should be greater than 6"}, 400
-    if not phone_number.startswith('+254'):
-      return {"message": "Phone number must start with +254"}, 400
+    if passporturl:
+      checkpassporturl = secure_filename(passporturl.filename)
+      passporturl.save(os.path.join(app.config['UPLOAD_FOLDER'], checkpassporturl))
+    if not firstname:
+      return {"status":400,"message": "Please provide first name" }, 400
+    if not lastname:
+      return{"status":400,"message": "Please provide first name"}, 400
+    if not email:
+      return{"status":400,"message": "Please provide email"}, 400
+    if not phonenumber:
+      return{"status":400,"message": "Please provide phone number"}, 400
+    if not passporturl:
+      return{"status":400,"message": "Please provide passport"}, 400  
+    if not re.match('^(?=\S{6,20}$)(?=.*?\d)(?=.*?[a-z])(?=.*?[A-Z])(?=.*?[^A-Za-z\s0-9])' ,password):
+      return {"status":400,"message": "Invalid password"}, 400
+    if not len(phonenumber) == 13:
+      return{"status":400,"message": "Email must have 12 character"}, 400
+    if not phonenumber.startswith('+254') or not phonenumber[1:13].isdigit():
+      return {"status":400,"message": "Please provide a valid phone number, start with +254"}, 400
     if not re.match(r"^[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]*$", email):
-      return {"message": "Please provide a valid email."}, 400
+      return {"status":400,"message": "Please provide a valid email."}, 400
     for user in users:
-      if email in user.values() or phone_number in user.values():
-        return {"message": "Email or phonr_number already exists"}, 409
+      if email in user.values():
+        return {"status":400,"message": "Email already exists"}, 409
+      if phonenumber in user.values():
+        return {"status":400,"message": "Phone number already exists"}, 409
```

Comment:
There is a conflict in the status codes here `400` and `409`, consider making the 2 same